### DISCUSSION TITLE
ZodResolver is flattening the nested objects in the zod schema.

### DISCUSSION THREAD

**Discussion by KarthikRaju391** [react-hook-form/react-hook-form#11576] [General]
The only thing is that I am not manually defining the zod schema. Instead, I am generating a zod schema automatically from a typescript object.

When I am parsing it manually, it's working correctly. 
```ts
	console.log(
		FormSchema.parse({
			title: 'test',
			overview: {
				metabar: {
					metabarTitle: 'met title',
				},
				description: getEmptyEditorState(),
			},
		})
	)
```

When the FormSchema is passed to the zodResolver, and I run `form.getValues()`, I am getting this
```ts
{
   description: undefined
   metabarTitle: undefined
   title: ""
}
```

It is not keeping the nested object fields like `overview` and `metabar`

**Comment by PieterT2000** [comment]
Bump
